src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,456 @@
+import static org.openkilda.model.PathComputationStrategy.LATENCY;
+import static org.openkilda.model.PathComputationStrategy.MAX_LATENCY;
+
+import org.openkilda.messaging.info.network.PathValidationResult;
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.Isl;
+import org.openkilda.model.IslStatus;
+import org.openkilda.model.PathSegment;
+import org.openkilda.model.PathValidationData;
+import org.openkilda.model.SwitchProperties;
+import org.openkilda.persistence.repositories.FlowRepository;
+import org.openkilda.persistence.repositories.IslRepository;
+import org.openkilda.persistence.repositories.SwitchPropertiesRepository;
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PathValidator {
+
+    private final IslRepository islRepository;
+    private final FlowRepository flowRepository;
+    private final SwitchPropertiesRepository switchPropertiesRepository;
+
+    private final SwitchRepository switchRepository;
+
+    public PathValidator(IslRepository islRepository,
+                         FlowRepository flowRepository,
+                         SwitchPropertiesRepository switchPropertiesRepository,
+                         SwitchRepository switchRepository) {
+        this.islRepository = islRepository;
+        this.flowRepository = flowRepository;
+        this.switchPropertiesRepository = switchPropertiesRepository;
+        this.switchRepository = switchRepository;
+    }
+
+    /**
+     * Validates whether it is possible to create a path with the given parameters. When there are obstacles, this
+     * validator returns all errors found on each segment. For example, when there is a path 1-2-3-4 and there is
+     * no link between 1 and 2, no sufficient latency between 2 and 3, and not enough bandwidth between 3-4, then
+     * this validator returns all 3 errors.
+     *
+     * @param pathValidationData path parameters to validate.
+     * @return a response object containing the validation result and errors if any
+     */
+    public PathValidationResult validatePath(PathValidationData pathValidationData) {
+        Set<String> result = pathValidationData.getPathSegments().stream()
+                .map(segment -> executeValidations(
+                        new InputData(pathValidationData, segment),
+                        new RepositoryData(islRepository, flowRepository, switchPropertiesRepository, switchRepository),
+                        getValidations(pathValidationData)))
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
```

Comment:
There could be duplicates, but removing them is expected. The errors contain the segment information, so there won't be removal of error from different segments. There could be exactly the same errors because they are given by different validators. For example, if there is no link on a segment, then you'll find it when validating bandwidth on this segment as well as latency on this segment.